failure in thread/test_async.timer_unset_from_handler/0

```
01:50:02 [ RUN      ] thread/test_async.timer_unset_from_handler/0
01:50:02 [1511999402.796741] [hpc-test-node3:61209:0]          async.c:330  UCX  WARN  releasing async context with 1 handlers
01:50:02 /scrap/jenkins/workspace/hpc-ucx-pr/label/hpc-test-node3/worker/3/contrib/../test/gtest/common/test.cc:242: Failure
01:50:02 Failed
01:50:02 Got 1 warnings during the test
01:50:02 
[  FAILED  ] thread/test_async.timer_unset_from_handler/0, where GetParam() = 1 (86 ms)
...
01:51:57 [----------] Global test environment tear-down
01:51:57 [==========] 1931 tests from 475 test cases ran. (1434334 ms total)
01:51:57 [  PASSED  ] 1930 tests.
01:51:57 [  FAILED  ] 1 test, listed below:
01:51:57 [  FAILED  ] thread/test_async.timer_unset_from_handler/0, where GetParam() = 1
01:51:57 
01:51:57  1 FAILED TEST
01:51:57 
make: *** [test] Error 1
```

--- a/test/gtest/ucs/test_async.cc
+++ b/test/gtest/ucs/test_async.cc
@@ -537,8 +537,12 @@ protected:
 
 UCS_TEST_P(test_async, timer_unset_from_handler) {
     local_timer_remove_handler lt(GetParam());
-    suspend_and_poll(&lt, COUNT);
+    ucs_time_t deadline = ucs_get_time() + ucs_time_from_sec(10.0);
+    do {
+        suspend_and_poll(&lt, 1);
+    } while ((lt.count() == 0) && (ucs_get_time() < deadline));
     EXPECT_GE(lt.count(), 1);
+    suspend_and_poll(&lt, COUNT);
     EXPECT_LE(lt.count(), 5); /* timer could fire multiple times before we remove it */
     int count = lt.count();
     suspend_and_poll(&lt, COUNT);

